### PR TITLE
Fix Qwen2.5-Omni audio preprocessing edge cases

### LINKED ISSUE #4458: [bugs report] Qwen2.5-Omni attention mask boundary and silent fbank can produce -inf
## Summary

I found two small audio preprocessing / encoder issues while testing Qwen2.5-Omni streaming audio in MNN.

1. `whisper_fbank()` takes `log(mel_specgram)` without clamping zero mel bins. Silent or all-zero tails can therefore produce `-inf` features.
2. In `Omni::audioProcess`, the Qwen2.5-Omni attention mask boundary list does not append the final boundary when `seqlen` is an exact multiple of `n_window`.

For a 2s Qwen2.5-Omni chunk:

- fbank frames = 200
- post-conv `seqlen = UP_DIV(200, 2) = 100`
- `n_window = 100`
- current `cu_seqlens` becomes `[0]`, not `[0, 100]`
- the loop that opens valid attention blocks never runs, so the whole block stays filled with `numeric_limits<float>::lowest()`

This makes exact-window streaming chunks produce bad audio embeddings.

## Suggested fix

In `tools/audio/source/audio.cpp`:

```cpp
mel_specgram = _Maximum(mel_specgram, _Scalar<float>(1e-10f));
auto log_specgram = _Log(mel_specgram) / _Log(_Scalar<float>(10.0));
```

In `transformers/llm/engine/src/omni.cpp`:

```cpp
if (cu_seqlens.empty() || cu_seqlens.back() != seqlen) {
    cu_seqlens.push_back(seqlen);
}
```

I also added a local regression test for 2s all-zero audio to check that `whisper_fbank()` outputs shape `[1,128,200]` and all finite values.

## Validation

With Qwen2.5-Omni-3B-MNN on a 34-case clean 5s boundary-word benchmark, each sample was split into non-overlap 2s chunks and encoded independently before embedding merge.

Before the mask fix:

- WER median: 1.091
- boundary recall median: 0.0

After the fbank clamp + boundary fix:

- 34/34 OK
- WER median: 0.091
- WER mean: 0.059
- boundary recall median: 1.0
- boundary recall mean: 0.935

Silence fbank regression:

- 2s all-zero waveform output shape `[1,128,200]`
- all values finite
- mean approximately `-1.5`

I have a minimal local patch with 3 files changed, 18 insertions, and 1 deletion.

### PR DESCRIPTION
**Pull request by Fatemanx**
## Summary
This PR fixes two Qwen2.5-Omni audio edge cases reported in #4458.

- Clamp Whisper mel bins before `log10` so silent / all-zero chunks do not produce `-inf` features.
- Fix exact-window Omni audio attention-mask boundaries so the final `seqlen` boundary is always present.
- Mirror the same boundary fix in the Python reference path.
- Add focused regressions for silent Whisper fbanks and exact-window Omni audio masks.

## Details
- `tools/audio/source/audio.cpp`
  - apply `max(mel_specgram, 1e-10)` before log in `whisper_fbank()`
- `transformers/llm/engine/src/omni.cpp`
  - build Omni audio window boundaries through a shared helper so boundaries always start at `0` and end at `seqlen`
- `transformers/llm/export/utils/audio.py`
  - keep Python boundary handling aligned with the C++ runtime
- `tools/audio/test/audio_test.cpp`
  - add a 2s all-zero regression that checks `[1, 128, 200]` output shape and finite values
- `transformers/llm/engine/test/test_omni_audio_window.cpp`
  - add exact-window boundary and mask-layout regression coverage

## Verification
- Built and ran `test_omni_audio_window`
- Built and ran `audio_test --gtest_filter=whisper_fbank.silence_is_finite`
- Also validated 2s zero-waveform Whisper fbanks produce shape `[1, 128, 200]` with all finite values in a focused local checker

Closes #4458

### CODE REVIEW

File: transformers/llm/engine/src/omni_audio_utils.hpp [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#ifndef OMNI_AUDIO_UTILS_HPP
+#define OMNI_AUDIO_UTILS_HPP
+
+#include <vector>
+
+namespace MNN {
+namespace Transformer {
+
+std::vector<int> buildOmniAudioWindowBoundaries(int seqlen, int n_window);
```
Comment (wangzhaode) [inline transformers/llm/engine/src/omni_audio_utils.hpp]: You can delete this file and its corresponding unit tests, and place this function directly into `omni.cpp`.

File: transformers/llm/engine/CMakeLists.txt [hunk 2]
```diff
@@ -99,6 +99,9 @@ if(MNN_LLM_BUILD_TEST)
     add_executable(test_tokenizer ${CMAKE_CURRENT_LIST_DIR}/test/test_tokenizer.cpp)
     target_include_directories(test_tokenizer PRIVATE ${CMAKE_CURRENT_LIST_DIR}/src/)
     target_link_libraries(test_tokenizer llm ${LLM_DEPS})
+    add_executable(test_omni_audio_window ${CMAKE_CURRENT_LIST_DIR}/test/test_omni_audio_window.cpp)
+    target_include_directories(test_omni_audio_window PRIVATE ${CMAKE_CURRENT_LIST_DIR}/src/)
+    target_link_libraries(test_omni_audio_window llm ${LLM_DEPS})
```
Comment (wangzhaode) [inline transformers/llm/engine/CMakeLists.txt]: remove this target also.

**Review by wangzhaode — CHANGES_REQUESTED** [review]
Hi @Fatemanx, thanks for the fix! A couple of changes needed before merging — please see the inline comments. Mainly: move the audio utility function directly into omni.cpp instead of creating a separate header, and remove the corresponding CMake target. Thanks!

File: transformers/llm/engine/src/omni_audio_utils.cpp [hunk 3]
```diff
@@ -0,0 +1,33 @@
+//
```
Comment (wangzhaode) [inline transformers/llm/engine/src/omni_audio_utils.cpp]: can delete this file and its corresponding unit tests, and place this function directly into omni.cpp.

**Comment by wangzhaode** [comment]
Hi @Fatemanx, friendly reminder — I've left a couple of inline review comments on this PR that need to be addressed:

1. **Move `omni_audio_utils.hpp` into `omni.cpp`** — please delete the separate header file and place the utility function directly in `omni.cpp` to keep things consolidated.
2. **Remove the corresponding CMake target** — clean up the build config accordingly.

Could you please update when you get a chance? If there's no response within a reasonable time, we may need to close this PR. Thanks for the contribution!

File: transformers/llm/engine/src/omni_audio_utils.hpp (hunk 1 above)
Comment (Fatemanx) [reply transformers/llm/engine/src/omni_audio_utils.hpp]: done.

File: transformers/llm/engine/CMakeLists.txt (hunk 2 above)
Comment (Fatemanx) [reply transformers/llm/engine/CMakeLists.txt]: done.

File: transformers/llm/engine/src/omni_audio_utils.cpp (hunk 3 above)
Comment (Fatemanx) [reply transformers/llm/engine/src/omni_audio_utils.cpp]: done.

**Comment by Fatemanx** [comment]
> Hi @Fatemanx, friendly reminder — I've left a couple of inline review comments on this PR that need to be addressed:
> 
> 1. **Move `omni_audio_utils.hpp` into `omni.cpp`** — please delete the separate header file and place the utility function directly in `omni.cpp` to keep things consolidated.
> 2. **Remove the corresponding CMake target** — clean up the build config accordingly.
> 
> Could you please update when you get a chance? If there's no response within a reasonable time, we may need to close this PR. Thanks for the contribution!

done.

**Review by wangzhaode — APPROVED** [review]
